cmd/apichecker/apicheck.go

```diff
@@ -0,0 +1,1418 @@
+						switch x.Val {
+						case "n4":
+							inner := tokenizer.Next()
+							if inner == html.TextToken {
+								if string(tokenizer.Text()) == "v2" {
+									case1 = true
+								}
+							}
+						case "n5":
+							inner := tokenizer.Next()
+							if inner == html.TextToken {
+								tempStr := string(tokenizer.Text())
+								if tempStr == "v3" {
+									case2 = true
+									resp = append(resp, tempStr)
+								}
+							}
+						case "n6":
+							inner := tokenizer.Next()
+							if inner == html.TextToken {
+								tempStr := string(tokenizer.Text())
+								if tempStr != "v4" {
+									case3 = true
+								}
+								if case1 && case2 && case3 {
+									return resp, nil
+								}
+								resp = append(resp, tempStr)
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+	return resp, nil
+}
+
+// HTMLScrapeLocalBitcoins gets the check string for Yobit Exchange
+func HTMLScrapeLocalBitcoins(htmlData *HTMLScrapingData) ([]string, error) {
+	temp, err := http.Get(htmlData.Path)
+	if err != nil {
+		return nil, err
+	}
+	a, err := ioutil.ReadAll(temp.Body)
+	if err != nil {
+		return nil, err
+	}
+	r, err := regexp.Compile(htmlData.RegExp)
+	if err != nil {
+		return nil, err
+	}
+	str := r.FindString(string(a))
+	sha := crypto.GetSHA256([]byte(str))
+	var resp []string
+	resp = append(resp, crypto.HexEncodeToString(sha))
+	return resp, nil
+}
+
+// TrelloCreateNewCard creates a new card on the list specified on trello
+func TrelloCreateNewCard(fillData *CardFill) error {
+	params := url.Values{}
+	params.Set("idList", fillData.ListID)
+	if fillData.Name != "" {
+		params.Set("name", fillData.Name)
+	}
+	if fillData.Desc != "" {
+		params.Set("desc", fillData.Desc)
+	}
+	if fillData.Pos != "" {
+		params.Set("pos", fillData.Pos)
+	}
+	if fillData.Due != "" {
+		params.Set("due", fillData.Due)
+	}
+	if fillData.MembersID != "" {
+		params.Set("idMembers", fillData.MembersID)
+	}
+	if fillData.LabelsID != "" {
+		params.Set("idLabels", fillData.LabelsID)
+	}
+	err := SendAuthReq(http.MethodPost,
+		pathNewCard+params.Encode()+apiKey+apiToken,
+		nil)
+	return err
+}
+
+// TrelloCreateNewCheck creates a new checklist item within a given checklist from trello
+func TrelloCreateNewCheck(newCheck string) error {
+	params := url.Values{}
+	params.Set("name", newCheck)
+	err := SendAuthReq(http.MethodPost,
+		pathChecklists+trelloChecklistID+params.Encode()+apiKey+apiToken,
+		nil)
+	return err
+}
+
+// TrelloCheckBoardID gets board id of the trello boards for a user which can be used for checking if a given board exists
+func TrelloCheckBoardID(username string) (bool, error) {
+	var data []MembersData
+	err := SendAuthReq(http.MethodGet,
+		fmt.Sprintf(pathCheckBoardID, username, apiKey, apiToken),
+		&data)
+	if err != nil {
+		return false, err
+	}
+	if data == nil {
+		return false, errors.New("no trello boards available for the given apikey and token")
+	}
+	for x := range data {
+		if trelloBoardID == data[x].ID {
```

Comment:
It would be more user friendly to support both friendly name ID's (data[x].Name) and trello alphanumeric ID's (data[x].ID)